src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
```

Comment:
The `MachineServiceImpl` class is parameterized with `Machine`, but it should be parameterized with a generic type `T` that extends `Machine` to handle different types of machines more flexibly.
```suggestion
public class MachineServiceImpl implements MachineService<T extends Machine> {
```